crates/wasmtime/src/runtime/instance.rs

```diff
@@ -452,6 +454,31 @@ impl Instance {
         self._get_export(store, entity, export_name_index)
     }
 
+    pub fn get_stack_pointer(&self, mut store: impl AsContextMut) -> Result<i32, ()> {
+        if let Some(sp_extern) = self.get_export(store.as_context_mut(), "__stack_pointer") {
```

Comment:
does this have to be nested like this?

Reply:
Updated, there is one unnecessary Err(()). Except this, all other Err(()) has its own error implication.